packages/theme/components/context/useConfig.tsx

```diff
@@ -0,0 +1,35 @@
+import React, {createContext, PropsWithChildren, useContext} from 'react'
+import type {PageMapItem} from 'nextra'
+
+export type ConfigContextLink = {
+  title: string
+  href: string
+  isActive?: boolean
+  isExternal?: boolean
+}
+
+export type ConfigContextValue = {
+  pageMap: PageMapItem[]
```

Comment:
Do we need to store pageMap in context? It has potential to be a huge object, and is also meant to be immutable and specifically not configurable, so struggling to see the use-case for it being part of a 'config'-oriented context.

Reply:
We don't _need_ to store it in context, but the previous set-up involved a lot of prop-drilling to get `pageMap` where it needed to be.

If you'd prefer the prop-drilling approach then I can remove it from context, but personally storing something as global as `pageMap` in a context feels cleaner to me than manually passing it down through components.

If you'd prefer me to pull it out into its own `PageMapContext` then I can, or I could rename the `ConfigContext` to something like `GlobalContext` or `DoctocatContext` or something

Reply:
> personally storing something as global as pageMap in a context feels cleaner to me

Yeah, can definitely see the argument for that. 

I think the overhead isn't worth it personally. It's not the kind of stateful data that justifies creating a dedicated context for. It looks like you're only using it in 2 different places right now too. Not much. FWIW, I also recall RSC best practice is to now prefer passing data through props over context if you can help it, as the latter is optimized for client rendering not server. For these reasons, let's take it out of the config context (as it's not a configurable data type anyway), and reconsider adding a separate context at a later date when / if that makes sense. That okay?

Reply:
It's the same problem for Primer Brand, yeah. Arguably less severe because Primer Brand components are purely presentational, so can be deferred to client or SSR.  pageMap is more akin to application data however, and something we would ideally run through the server.